### PR TITLE
Move properties returned only by single method

### LINKED ISSUE #2: Distinguish between types and return types
Some properties are "returned only in `getChat`" or "...`getMe`". They should be moved from the base interfaces and put only in the return value they're present in.

Example:
```js
interface User { /* ... */ } // e.g. used in interface ChatMember
interface GottenUser extends User { /* ... */ } // used exclusively in getMe
```

### PR DESCRIPTION
**Pull request by wojpawlik**
https://github.com/telegraf/telegraf/pull/1115#issuecomment-679037328

Fixes #2

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
Would you mind moving the added properties to new interfaces as outlined in the issue?

Taken from [the docs](https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases):
> we recommend using an interface over a type alias when possible

File: methods.d.ts [hunk 1]
```diff
@@ -89,7 +90,14 @@ export interface Telegram {
...
   /** A simple method for testing your bot's auth token. Requires no parameters. Returns basic information about the bot in form of a User object. */
-  getMe(): User;
+  getMe(): User & {
+    /** True, if the bot can be invited to groups. Returned only in getMe. */
+    can_join_groups: Boolean;
+    /** True, if privacy mode is disabled for the bot. Returned only in getMe. */
+    can_read_all_group_messages: Boolean;
+    /** True, if the bot supports inline queries. Returned only in getMe. */
+    supports_inline_queries: Boolean;
+  };
```
Comment (KnorpelSenf) [inline methods.d.ts:100]: Please move to interface declaration

File: methods.d.ts [hunk 2]
```diff
@@ -657,7 +665,24 @@ export interface Telegram {
...
+    pinned_message?: Message;
+    /** Default chat member permissions, for groups and supergroups. Returned only in getChat. */
+    permissions?: ChatPermissions;
+    /** For supergroups, the minimum allowed delay between consecutive messages sent by each unpriviledged user. Returned only in getChat. */
+    slow_mode_delay?: Integer;
+    /** For supergroups, name of group sticker set. Returned only in getChat. */
+    sticker_set_name?: String;
+    /** True, if the bot can change the group sticker set. Returned only in getChat. */
+    can_set_sticker_set?: Boolean;
+  };
```
Comment (KnorpelSenf) [inline methods.d.ts]: Please move to interface declaration

File: types.d.ts [hunk 3]
```diff
@@ -83,20 +86,23 @@ export interface Chat {
...
+  pinned_message: Message;
   /** Default chat member permissions, for groups and supergroups. Returned only in getChat. */
-  permissions?: ChatPermissions;
+  permissions: ChatPermissions;
   /** For supergroups, the minimum allowed delay between consecutive messages sent by each unpriviledged user. Returned only in getChat. */
-  slow_mode_delay?: Integer;
+  slow_mode_delay: Integer;
   /** For supergroups, name of group sticker set. Returned only in getChat. */
-  sticker_set_name?: String;
+  sticker_set_name: String;
```
Comment (wojpawlik) [inline types.d.ts]: I'm not sure these should be marked as required

**Comment by wojpawlik** [comment]
LGTM; please squash and merge.

**Comment by wojpawlik** [comment]
That was not a squash :)

Your GitHub name (not username) and email should match your `git config user.name` and `git config user.email`, respectively.

One more idea, generate type guards from `types.d.ts`. I'll PR if I find a suitable generator.

**Comment by KnorpelSenf** [comment]
That's true, I hadn't seen your comment before merging.

I'm using a few different emails and I sometimes I just don't set local git config after opening a new repo. Does that cause issues?

**Comment by wojpawlik** [comment]
Yes, `git` sees you as 2 different people:

```
$ git shortlog
KnorpelSenf (2):
      Initial commit
      Merge pull request #1 from wojpawlik/only     

Steffen (20):
      Add specs
      Add .editorconfig
      Comment reformatting, consistently use aliases
```

You can use conditional config to set email based on directory for projects you haven't committed to yet, or set up [`.mailmap`](https://www.git-scm.com/docs/git-check-mailmap#_mapping_authors).

**Pull request merged**